src/Compiler/Service/ItemKey.fs

```diff
@@ -416,8 +416,16 @@ and [<Sealed>] ItemKeyStoreBuilder() =
             writeString ItemKeyTags.itemDelegateCtor
             writeType false ty
 
+        // Named argument in a signature
+        | Item.OtherName (ident = Some (ident); argType = ty; argInfo = Some _) ->
```

Comment:
Yes agreed it looks like we need an ArgumentContainer here

Reply:
The container is attached to the Item, and my understanding is that at the point we create the Item.OtherName we have the necessary Val to hand and can just include it.  So it should really just be a matter of adding a Val or ValRef case to the ArgumentContainer type, rather than further modifying the TypedTree.

Reply:
The container is attached to `Item.OtherName`, but not to `Item.Value` which we need to match with it (the arguments coming from checking lambdas).

Reply:
Another way might be to write either the definition or signature range here. Unfortunately it's not always available (`OtherRange` can still be `None` here even though we set it in signature conformance).